examples/c/5_series/gnss_ins/5_series_gnss_ins_example.c

```diff
@@ -1,354 +1,1106 @@
+    const uint16_t minor = device_info.firmware_version / 100 % 10;
+    const uint16_t patch = device_info.firmware_version % 100;
+
+    // Firmware version format is x.x.xx
+    char firmwareVersion[16];
+    snprintf(firmwareVersion, sizeof(firmwareVersion) / sizeof(firmwareVersion[0]), "%d.%d.%02d",
+        major,
+        minor,
+        patch
+    );
+
+    MICROSTRAIN_LOG_INFO("-------- Device Information --------\n");
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Name", device_info.model_name);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Model Number", device_info.model_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Serial Number", device_info.serial_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Lot Number", device_info.lot_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Options", device_info.device_options);
+    MICROSTRAIN_LOG_INFO("%-16s | %16s\n", "Firmware Version", firmwareVersion);
+    MICROSTRAIN_LOG_INFO("------------------------------------\n");
+
+    // Load the default settings on the device
+    // Note: This guarantees the device is in a known state
+    MICROSTRAIN_LOG_INFO("Loading default settings.\n");
+    cmd_result = mip_3dm_default_device_settings(_device);
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not load device default settings!\n");
+    }
+}
+
 ////////////////////////////////////////////////////////////////////////////////
+/// @brief Handles graceful program termination and cleanup
+///
+/// @details Handles graceful shutdown when errors occur:
+///          - Outputs provided error message
+///          - Closes device connection if open
+///          - Exits with appropriate status code
+///
+/// @param _device_port Serial port connection to close
+/// @param _message Error message to display
+/// @param _successful Whether termination is due to success or failure
+///
+void terminate(serial_port* _device_port, const char* _message, const bool _successful)
+{
+    if (strlen(_message) != 0)
+    {
+        if (_successful)
+        {
+            MICROSTRAIN_LOG_INFO("%s", _message);
+        }
+        else
+        {
+            MICROSTRAIN_LOG_ERROR("%s", _message);
+        }
+    }
+
+    if (_device_port == NULL)
+    {
+        // Initialize the device interface with a serial port connection
+        MICROSTRAIN_LOG_ERROR("Connection not set for the device interface. Cannot close the connection.\n");
+    }
+    else
+    {
+        if (serial_port_is_open(_device_port))
+        {
+            MICROSTRAIN_LOG_INFO("Closing the connection.\n");
+
+            if (!serial_port_close(_device_port))
+            {
+                MICROSTRAIN_LOG_ERROR("Failed to close the connection!\n");
+            }
+        }
+    }
 
-bool should_exit()
+    MICROSTRAIN_LOG_INFO("Exiting the program.\n");
+
+#ifdef _WIN32
+    // Keep the console open on Windows
+    system("pause");
```

Comment:
Is this still needed for modern MSVC? Also what if they're using mingw - should this be `_MSVC` or similar instead?